GliaWidgetsTests/SecureConversations/ChatTranscript/TranscriptModelTests+GVA.swift

```diff
@@ -0,0 +1,113 @@
+@testable import GliaWidgets
+import XCTest
+
+extension SecureConversationsTranscriptModelTests {
+    func test_gvaLinkButtonAction() {
+        let options: [GvaOption] = [
+            .mock(url: "http://mock.mock"),
+            .mock(url: "https://mock.mock"),
+            .mock(url: "mock://mock.self", urlTarget: "self"),
+            .mock(url: "mock://mock.modal", urlTarget: "modal"),
+            .mock(url: "mock://mock", urlTarget: "mock"),
+            .mock(url: "mailto:mock@mock.mock"),
+            .mock(url: "tel:12345678")
+        ]
+        var calls: [Call] = []
+
+        let viewModel = createViewModel()
+        viewModel.environment.uiApplication.open = { url in
+            calls.append(.openUrl(url.absoluteString))
+        }
+        viewModel.environment.sendSecureMessage = { message, attachment, _, _ in
+            calls.append(.sendOption(message, attachment?.selectedOption))
+            return .mock
+        }
+        options.forEach {
+            viewModel.gvaOptionAction(for: $0)()
+        }
+
+        let expectedResult: [Call] = [
+            .openUrl("http://mock.mock"),
+            .openUrl("https://mock.mock"),
+            .openUrl("mock://mock.self"),
+            .openUrl("mock://mock.modal"),
+            .openUrl("mailto:mock@mock.mock"),
+            .openUrl("tel:12345678")
+        ]
+        XCTAssertEqual(calls, expectedResult)
+    }
+
+    func test_gvaPostbackButtonAction() {
+        let option = GvaOption.mock(text: "text", value: "value")
+        var calls: [Call] = []
+        let viewModel = createViewModel()
+        // To ensure `open` is not called in case of URL Button
+        viewModel.environment.uiApplication.open = { url in
+            calls.append(.openUrl(url.absoluteString))
+        }
+        viewModel.environment.sendSecureMessage = { message, attachment, _, _ in
+            calls.append(.sendOption(message, attachment?.selectedOption))
+            return .mock
+        }
+        viewModel.gvaOptionAction(for: option)()
+
+        XCTAssertEqual(calls, [.sendOption("text", "value")])
+    }
+
+    func test_broadcastEventAction() {
+        let option = GvaOption.mock(text: "text", destinationPdBroadcastEvent: "mock")
+        var calls: [Call] = []
+        let viewModel = createViewModel()
+        viewModel.engagementAction = { action in
+            guard case .showAlert = action else { return }
+            calls.append(.showAlert)
+        }
+
+        viewModel.gvaOptionAction(for: option)()
+
+        XCTAssertEqual(calls, [.showAlert])
+    }
+}
+
+private extension SecureConversationsTranscriptModelTests {
+    enum Call: Equatable {
+        case openUrl(String?)
+        case sendOption(String?, String?)
+        case showAlert
+
+        static func == (lhs: Call, rhs: Call) -> Bool {
```

Comment:
`Equatable` can be synthesized by compiler.  Is there something specific in this `==` implementation?

Reply:
You're right. Good catch!
Fixed b535b617199a51a3e7a3f0152d46279ade0dff9f